core/src/components/input/test/shape/input.e2e.ts

```diff
@@ -109,3 +109,60 @@ configs({ modes: ['ionic-md', 'md'] }).forEach(({ title, screenshot, config }) =
     });
   });
 });
+
+// Rectangular shape is only available in ionic theme
+// TODO(FW-6098): Add test for rectangular shape in md
+// by combining these tests with the above tests
+configs({ modes: ['ionic-md'] }).forEach(({ title, screenshot, config }) => {
+  test.describe(title('input: shape'), () => {
+    test.describe('rectangular shape', () => {
+      test.describe('outline fill', () => {
+        test('should not have visual regressions', async ({ page }) => {
+          await page.setContent(
+            `
+            <ion-input
+              shape="rectangular"
+              fill="outline"
+              label="Email"
+              value="hi@ionic.io"
+              helper-text="Enter your email"
+              maxlength="20"
+              counter="true"
+            ></ion-input>
+            `,
+            config
+          );
+
+          const input = page.locator('ion-input');
+          await expect(input).toHaveScreenshot(screenshot(`input-shape-rectangular-fill-outline`));
+        });
+
+        test('border radius should be customizable', async ({ page }) => {
```

Comment:
Do we need to add this test? I wonder if it's redundant to the one in outline.

Reply:
I don't think it's redundant since we use different selectors to style the border radius based on the shape. This ensures it still works with the rectangular shape if we end up changing the ionic theme to act like md in the future: https://github.com/ionic-team/ionic-framework/blob/704765e6203bc35c9ef78fabe6e15980bfa834ec/core/src/components/input/input.md.outline.scss#L200-L209